libraries/codegen-tooling/prebuild_hook.py

```diff
@@ -0,0 +1,198 @@
+"""Download and install script for github.com/uw-midsun/codegen-tooling
+
+Fetches the latest out codegen-tooling-out.zip release and unpacks it for
+the build system to pick up.
+"""
+#! /usr/bin/env python
+
+import subprocess
+import os
+import re
+import zipfile
+import shutil
+import socket
+
+# Fetch the latest release from the official GitHub API
+TAG = 'latest'
+URL = 'https://api.github.com/repos/uw-midsun/codegen-tooling/releases/'
+# This file name may need to be updated
+FILE = 'codegen-tooling-out.zip'
+
+# Hash the file to prevent reinstalling the same file.
+HASH_FILE = 'hash.txt'
+
+# For validating there is an internet connection
+REMOTE_SERVER = "www.google.com"
+
+
+def connected():
+    """Connected checks for a internet connection
+
+    Tries to connect to Google over an HTTP socket and will fail if there is no
+    connection available.
+    """
+    try:
+        host = socket.gethostbyname(REMOTE_SERVER)
+        _ = socket.create_connection((host, 80), 2)
+        return True
+    except Exception:
+        pass
+    return False
+
+
+def fetch_release(url, tag, output_file):
```

Comment:
Would it make sense to name the latest release archive as the hash name, so we potentially avoid downloading the ZIP each time?